src/file_io.c

```diff
@@ -794,33 +834,21 @@ psf_open_handle (PSF_FILE * pfile)
 	cfParams.dwSize = sizeof (CREATEFILE2_EXTENDED_PARAMETERS) ;
 	cfParams.dwFileAttributes = FILE_ATTRIBUTE_NORMAL ;
 
-	handle = CreateFile2 (pfile->path.wc, dwDesiredAccess, dwShareMode, dwCreationDistribution, &cfParams) ;
-
-	return handle ;
+	handle = CreateFile2 (pwszPath, dwDesiredAccess, dwShareMode, dwCreationDistribution, &cfParams) ;
 #else
-	if (pfile->use_wchar)
-		handle = CreateFileW (
-					pfile->path.wc,				/* pointer to name of the file */
-					dwDesiredAccess,			/* access (read-write) mode */
-					dwShareMode,				/* share mode */
-					0,							/* pointer to security attributes */
-					dwCreationDistribution,		/* how to create */
-					FILE_ATTRIBUTE_NORMAL,		/* file attributes (could use FILE_FLAG_SEQUENTIAL_SCAN) */
-					NULL						/* handle to file with attributes to copy */
-					) ;
-	else
-		handle = CreateFileA (
-					pfile->path.c,				/* pointer to name of the file */
-					dwDesiredAccess,			/* access (read-write) mode */
-					dwShareMode,				/* share mode */
-					0,							/* pointer to security attributes */
-					dwCreationDistribution,		/* how to create */
-					FILE_ATTRIBUTE_NORMAL,		/* file attributes (could use FILE_FLAG_SEQUENTIAL_SCAN) */
-					NULL						/* handle to file with attributes to copy */
-					) ;
+	handle = CreateFileW (
```

Comment:
btw, have you considered just dumping the wide interfaces and relying on Windows 10 1903+ UTF-8 support in the `*A` interfaces?

Reply:
I guess this feature is related to UWP applications.